source/lib/notifications-service.js

```diff
@@ -6,75 +6,119 @@ import localStore from './local-store';
 
 const syncStore = new OptionsSync();
 
-export const closeNotification = async notificationId => {
-	return browser.notifications.clear(notificationId);
-};
+function getLastReadForNotification(notification) {
+	// Extract the specific fragment URL for a notification
+	// This allows you to directly jump to a specif comment as if you were using
+	// the notifications page
+	const lastReadTime = notification.last_read_at;
+	const lastRead = new Date(lastReadTime || notification.updated_at);
+
+	if (lastReadTime) {
+		lastRead.setSeconds(lastRead.getSeconds() + 1);
+	}
 
-export const openNotification = async notificationId => {
-	const url = await localStore.get(notificationId);
+	return lastRead.toISOString();
+}
 
-	await closeNotification(notificationId);
+async function issueOrPRHandler(notification) {
+	const notificationUrl = notification.subject.url;
+
+	try {
+		const {pathname} = new URL(notificationUrl);
+		const lastRead = getLastReadForNotification(notification);
 
-	if (url) {
 		try {
-			const {json} = await makeApiRequest(new URL(url).pathname);
-			const targetUrl = json.message === 'Not Found' ? await getTabUrl() : json.html_url;
-			return openTab(targetUrl);
-		} catch (_) {
-			return openTab(await getTabUrl());
+			// Try to get the latest comment that the user has not read
+			const {json: comments} = await makeApiRequest(pathname + '/comments', {
+				since: lastRead,
+				per_page: 1 // eslint-disable-line camelcase
+			});
+
+			const comment = comments[0];
+			if (comment) {
+				return comment.html_url;
+			}
+
+			// If there are not comments or events, then just open the url
+			const {json: response} = await makeApiRequest(pathname);
+			const targetUrl = response.message === 'Not Found' ? await getTabUrl() : response.html_url;
+			return targetUrl;
+		} catch (error) {
+			return notificationUrl;
 		}
+	} catch (error) {
+		throw error;
 	}
-
-	return false;
+}
+
+const notificationHandlers = {
+	/* eslint-disable quote-props */
```

Comment:
if you're going to disable the rule, you might want to unquote the keys and disable whatever rule complains instead.

Reply:
Disabling `quote-props` would only trigger `camel-case` next, so had to choose one of them. 🤷‍♂️